kubevirtexternalinfra/externalinfra.go

```diff
@@ -251,3 +254,45 @@ func GetKubeConfig(ctx context.Context, cl client.Client, secretNamespace, secre
 
 	return kubeConfig, nil
 }
+
+func ValidateClusterVersions(ctx context.Context, cl KubevirtInfraClient) error {
+	var cnvVersion, k8sVersion *semver.Version
+
+	eg, egCtx := errgroup.WithContext(ctx)
+
+	eg.Go(func() error {
+		var err error
+		cnvVersion, err = cl.GetInfraKubevirtVersion(egCtx)
+		return err
+	})
+
+	eg.Go(func() error {
+		var err error
+		k8sVersion, err = cl.GetInfraK8sVersion()
+		return err
+	})
+
+	err := eg.Wait()
+	if err != nil {
+		return err
+	}
```

Comment:
that's kind of neat. I haven't seen this usage of errgroups before. looks good

Reply:
https://medium.com/@yardenlaif/go-sync-or-go-home-errgroup-f91a0ee72d3f